modules/cms/formwidgets/Components.php

```diff
@@ -42,7 +43,7 @@ protected function listComponents()
             try {
                 $componentObj = $manager->makeComponent($name, null, $properties);
 
-                $componentObj->alias = $alias;
+                $componentObj->alias = ((substr($name, 0, 1) === '@' && $alias !== $name) ? '@' : '') . $alias;
```

Comment:
Could you explain what this does exactly? Taking me too long to mentally parse

Reply:
This line determines if the component is a soft component (by being prefixed with an `@`) and if it is, and an alias is in use, it prefixes the alias with `@`. This is for the component property inspector - it allows people to add `@` to an alias to make any component a soft component, and ensures that the `@` is visible when the component properties are reloaded.